akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala

```diff
@@ -174,6 +174,30 @@ class FramingSpec extends AkkaSpec {
         3.seconds) should ===(Vector.empty)
     }
 
+    "work with grouped frames" in {
+
+      for {
+        byteOrder ← byteOrders
+        fieldOffset ← fieldOffsets
+        fieldLength ← fieldLengths
+        count ← 2 to frameLengths.count(_ < (1L << (fieldLength * 8)))
+      } {
+
+        val encodedFrames = frameLengths.filter(_ < (1L << (fieldLength * 8))).map { length ⇒
+          val payload = referenceChunk.take(length)
+          encode(payload, fieldOffset, fieldLength, byteOrder)
+        }
+        val grouped = encodedFrames.grouped(count).map(_.fold(ByteString.empty)((r, bs) ⇒ r ++ bs)).toList
+
+        Await.result(
+          Source(encodedFrames)
+            .via(Framing.lengthField(fieldLength, fieldOffset, Int.MaxValue, byteOrder))
+            .grouped(10000)
+            .runWith(Sink.head),
+          3.seconds) should ===(encodedFrames)
+      }
+    }
+
```

Comment:
The easiest way to trigger the regression you introduced would be to use `TestPublisher` and `TestSubscriber` here to manually signal demand and elements. You can look at `akka.stream.scaladsl.FlowBatchSpec` to see how to use those.

The test should fail until you have aligned the rewritten stage with the old one, so that's how you will know that you have succeeded.

To repeat the error case, upstream has pushed a chunk with one or more frames. Before any demand comes from downstream, upstream closes. Now downstream should be able to signal demand to consume each full frame, not until that is done and the buffer is empty it completes or there is one incomplete frame left, then it should fail with the warning.